### ISSUE TITLE
VerifyError on unclear overrides

### ISSUE THREAD

**Issue by martijnhoekstra** [scala/bug#11749]
Not-very-but-still-a-bit-minimized, on scala 2.12.10 and 2.12.8 (and probably elsewhere, but not tested)

```
import scala.collection.mutable.MapLike
import scala.collection.mutable.Map

abstract class Base extends Map[String, String]

abstract class ImplBase extends Base {
  final def get(key: String): Option[String] = Some("result")
  final def iterator: Iterator[(String, String)] = Iterator.empty
}
final class Leaf extends ImplBase with MapLike[String, String, Leaf] {
  override def empty: Leaf = new Leaf()
  
  def +=(kv: (String, String)): this.type = this
  def -=(key: String): this.type = this
  
}

new Leaf()
```

> java.lang.VerifyError: class Leaf overrides final method get.(Ljava/lang/Object;)Lscala/Option;

Apologies for not further mimizing (without stdlib). 

Workaround exists by removing the final modifier from `get`

**Comment by lrytz** [comment]
We did a lot of fixes in this area for 2.13, see https://github.com/scala/bug/issues/7917, https://github.com/scala/bug/issues/10853, PRs https://github.com/scala/scala/pull/7439, https://github.com/scala/scala/pull/7941, https://github.com/scala/scala/pull/7929.

I think it's quite likely that the issue is fixed in 2.13. We're probably not going to backport these changes to 2.12.

**Comment by martijnhoekstra** [comment]
Yes, it's fixed in 2.13, I should have said that in the original report. I can imagine you'll want to wont-fix this then.

**Comment by lrytz** [comment]
Yeah, I think so... Thanks for the report.
